lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""OpenSearch user helper functions.
+
+These functions wrap around some API calls used for user management.
+"""
+
+import logging
+from typing import Any, Dict, List, Optional
+
+from charms.opensearch.v0.opensearch_distro import OpenSearchDistribution
+
+logger = logging.getLogger(__name__)
+
+
+USER_ENDPOINT = "/_plugins/_security/api/internalusers"
+ROLE_ENDPOINT = "/_plugins/_security/api/roles"
+
+
+class OpenSearchUserMgmtError(Exception):
+    """Base exception class for OpenSearch user management errors."""
+
+
+class OpenSearchUserManager:
+    """User management class for OpenSearch API."""
+
+    def __init__(self, opensearch: OpenSearchDistribution):
+        self.opensearch = opensearch
+
+    def create_role(
+        self,
+        role_name: str,
+        permissions: Optional[Dict[str, str]],
+        action_groups: Optional[Dict[str, str]],
+    ) -> Dict[str, Any]:
```

Comment:
The return type must be replaced with `None` since it's no longer returning anything
(Same comment on all the other methods of this class)